### ISSUE TITLE
log4j2.xml: configuration value lookup does not work as expected (logging.server.retention-accumulated-to-keep)

### ISSUE THREAD

**Issue by saschaszott** [4Science/DSpace#320]
## Describe the bug

There are **five** configuration value lookups in `log4j2.xml` of the form

```xml
<IfAccumulatedFileCount exceeds='${logging.server.retention-accumulated-to-keep}'/>
```

The lookup of the configuration setting `logging.server.retention-accumulated-to-keep` fails in this case as the configuration bean is not available to Log4J.

Subsequently, every CLI command execution produces **five** Log4J error messages:

```
2023-03-01 16:56:40,625 main ERROR IfAccumulatedFileCount invalid or missing threshold value.
2023-03-01 16:56:40,640 main ERROR IfAccumulatedFileCount invalid or missing threshold value.
2023-03-01 16:56:40,642 main ERROR IfAccumulatedFileCount invalid or missing threshold value.
2023-03-01 16:56:40,644 main ERROR IfAccumulatedFileCount invalid or missing threshold value.
2023-03-01 16:56:40,646 main ERROR IfAccumulatedFileCount invalid or missing threshold value.
```

I've checked that my DSpace configuration file `dspace.cfg` contains a setting

```properties
logging.server.retention-accumulated-to-keep = 1000
```

##  Steps To Reproduce

1. run any CLI command, e.g. `$DSHOME/dspace/bin/dspace database info`
2. 5 error messages are generated due to failed lookup of `logging.server.retention-accumulated-to-keep`

## Expected behavior

Configuration lookup is successful and no error messages are generated.

## Related work

The changes in `log4j2.xml` were added by this commit https://github.com/4Science/DSpace/commit/ac7ea17f971b5d2dfd9f26b52c3d5fe561872b81#diff-39220e4ec26d2bc963fa3162bda3fbe3a9076354cb73efc5525c89f57033a1f6

**Comment by floriangantner** [comment]
@saschaszott it might fix your problem if you add `<Property name="logging.server.retention-accumulated-to-keep">${logging.server.retention-accumulated-to-keep}</Property>` to the `Properties` Element in log4j2.xml

**Comment by saschaszott** [comment]
Hi @floriangantner , if have added 

```xml
<Property name="logging.server.retention-accumulated-to-keep">${logging.server.retention-accumulated-to-keep}</Property>
```

to the `Properties` section of `log4j2.xml`. The (five) error messages still appear. I'm not sure if the DSpace configuration context is accessible to Log4J?

**Comment by corrad82-4s** [comment]
configuration context should be accessible to Log4J, is this property in some way overridden by your instance `local.cfg` file?

**Comment by saschaszott** [comment]
> configuration context should be accessible to Log4J, is this property in some way overridden by your instance `local.cfg` file?

No, my `local.cfg` does not contain `logging.server.retention-accumulated-to-keep`.

**Comment by corrad82-4s** [comment]
In current `dspace-cris-7` main branch, going to be released, this issue has been solved. 
Now log4j2.xml file in installation `config` directory has the `IfAccumulatedFileCount` property set with value of configuration property `logging.server.retention-accumulated-to-keep`. 
Closing this issue, should something go not as expected, feel free to reopen it.

**Comment by saschaszott** [comment]
The change in `log4j2.xml` is **not** contained in the "main" branch (`dspace-cris-7`). The change in `log4j2.xml` was committed to the branch `CST-10158` by this commit https://github.com/4Science/DSpace/commit/78540199858fb650176cb9695a58de8ec73706df
It was not merged into the `dspace-cris-7` branch. 
@corrad82-4s , could you please reopen this issue?

**Comment by corrad82-4s** [comment]
as discussed with you @saschaszott , issue seems not resolved 
> The change in log4j2.xml is not contained in the "main" branch (dspace-cris-7)

we need to look into it as per our tests injection of properties, from dspace configuraiton, in log4j configuration file was working as expected.

**Comment by saschaszott** [comment]
@abollini , this problem still exists in DSC 2024.02.00

```
root@17f28ef8837c:/dspace/bin# ./dspace version
2025-01-03T10:51:56.392958592Z main ERROR IfAccumulatedFileCount invalid or missing threshold value.
2025-01-03T10:51:56.411531509Z main ERROR IfAccumulatedFileCount invalid or missing threshold value.
2025-01-03T10:51:56.416289009Z main ERROR IfAccumulatedFileCount invalid or missing threshold value.
2025-01-03T10:51:56.421348926Z main ERROR IfAccumulatedFileCount invalid or missing threshold value.
2025-01-03T10:51:56.423638051Z main ERROR IfAccumulatedFileCount invalid or missing threshold value.
DSpace version:  cris-2024.02.00
  SCM revision:  UNKNOWN_REVISION
    SCM branch:  UNKNOWN_BRANCH
            OS:  Linux(aarch64) version 6.10.14-linuxkit
  Applications:
     Discovery:  enabled.
           JRE:  Eclipse Adoptium version 17.0.13
   Ant version:  null
 Maven version:  3.9.9
   DSpace home:  /dspace
```
